colcon_in_container/providers/lxd.py

```diff
@@ -67,120 +69,165 @@ def __init__(self, ros_distro, pro_token=None):  # noqa: D107
             cloud_init_url = \
                 'https://cloud-images.ubuntu.com/minimal/releases/'
 
-        config: Dict[str, Any] = {
-            'name': self.instance_name,
-            'source': {
-                'type': 'image',
-                'protocol': 'simplestreams',
-                'server': cloud_init_url,
-                'alias': f'{self.ubuntu_distro}/'
-                f'{host_architecture()}',
-            },
-            'ephemeral': True,
-            'config': {},
-        }
-        config['config']['user.user-data'] \
-            = self._render_jinja_template(pro_token)
-
-        if self.lxd_client.instances.exists(self.instance_name):
-            previous_instance = self.lxd_client.instances.get(
-                self.instance_name)
-            if previous_instance.status == 'Running':
-                previous_instance.stop(wait=True)
-            else:
-                previous_instance.delete(wait=True)
+        image_alias = f'{self.ubuntu_distro}/{host_architecture()}'
+        cloud_init_data = self._render_jinja_template(pro_token)
+
+        # Check if instance already exists and clean it up
+        if self._instance_exists(self.instance_name):
+            instance_status = self._get_instance_status(self.instance_name)
+            if instance_status == 'Running':
+                self._stop_instance(self.instance_name)
+            # Delete the instance after stopping
+            self._delete_instance(self.instance_name)
 
         logger.info('Downloading the image then creating the LXD instance')
-        self.instance = self.lxd_client.instances.create(config, wait=True)
-        self.instance.start(wait=True)
+
+        # Create instance (but don't start it yet)
+        subprocess.run([
+            'lxc', 'init',
+            f'{cloud_init_url}:{image_alias}',
```

Comment:
You don't need to specify the cloud init here but simply using "ubuntu-minimal".